src/protocol/mod.rs

```diff
@@ -123,6 +124,45 @@ macro_rules! repeat64str {
     }
 }
 
+/// Bounds-checked index type for protocol steps indexed 0..64.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
```

Comment:
how do you keep them sorted? do you have to do it manually or there is a magic cargo command I could use to have the same result?

Reply:
When I last looked at this, I discovered that ordering matters.  That is, `cargo fmt` wasn't able to reorder these.  (I'm surprised that Eq and PartialEq can appear in this order, but maybe my info is old.)

Reply:
I just happened to write it in sorted order without thinking about Eq / PartialEq, I guess that must have changed recently. There are a few places in the code that list them in Eq, PartialEq order now.

Maybe cargo fmt will start sorting them at some point.

Reply:
Yeah, they must have worked out how to address the ordering problem in the compiler, which is pretty good if it means we can eliminate variance at some point in the future.

Reply:
@martinthomson can you point me to this issue if you have it handy? I am curious what could be the problem.